tests/search/test-class-search.php

```diff
@@ -1117,6 +1147,336 @@ public function test__query_count_incr() {
 		}
 	}
 
+	public function test__is_query_mirroring_enabled_no_constant_no_option() {
+		$es = new \Automattic\VIP\Search\Search();
+		
+		$enabled = $es->is_query_mirroring_enabled();
+
+		$this->assertFalse( $enabled );
+	}
+
+	/**
+	 * @runInSeparateProcess
+	 * @preserveGlobalState disabled
+	 */
+	public function test__is_query_mirroring_enabled_via_option() {
+		update_option( 'vip_enable_search_query_mirroring', true );
+
+		$es = new \Automattic\VIP\Search\Search();
+		
+		$enabled = $es->is_query_mirroring_enabled();
+
+		delete_option( 'vip_enable_search_query_mirroring' );
+
+		$this->assertTrue( $enabled );
+	}
+
+	/**
+	 * @runInSeparateProcess
+	 * @preserveGlobalState disabled
+	 */
+	public function test__is_query_mirroring_enabled_with_option_false() {
+		update_option( 'vip_enable_search_query_mirroring', false );
+
+		$es = new \Automattic\VIP\Search\Search();
+		
+		$enabled = $es->is_query_mirroring_enabled();
+
+		delete_option( 'vip_enable_search_query_mirroring' );
+
+		$this->assertFalse( $enabled );
+	}
+
+	/**
+	 * @runInSeparateProcess
+	 * @preserveGlobalState disabled
+	 */
+	public function test__is_query_mirroring_enabled_via_constant() {
+		define( 'VIP_ENABLE_SEARCH_QUERY_MIRRORING', true );
+
+		$es = new \Automattic\VIP\Search\Search();
+		
+		$enabled = $es->is_query_mirroring_enabled();
+
+		$this->assertTrue( $enabled );
+	}
+
+	/**
+	 * @runInSeparateProcess
+	 * @preserveGlobalState disabled
+	 */
+	public function test__is_query_mirroring_enabled_with_constant_false() {
+		define( 'VIP_ENABLE_SEARCH_QUERY_MIRRORING', false );
+
+		$es = new \Automattic\VIP\Search\Search();
+		
+		$enabled = $es->is_query_mirroring_enabled();
+
+		$this->assertFalse( $enabled );
+	}
+
+	/**
+	 * @runInSeparateProcess
+	 * @preserveGlobalState disabled
+	 */
+	public function test__should_mirror_wp_query_when_mirroring_disabled() {
+		define( 'VIP_ENABLE_SEARCH_QUERY_MIRRORING', false );
+
+		$es = new \Automattic\VIP\Search\Search();
+
+		$query = new \WP_Query();
+
+		$should_mirror = $es->should_mirror_wp_query( $query );
+
+		$this->assertFalse( $should_mirror );
+	}
```

Comment:
Which makes all the `should_mirror_wp_query` tests pass because you only `assertFalse` against them.